library/Ivoz/Provider/Domain/Model/Friend/Friend.php

```diff
@@ -228,6 +229,21 @@ public function setRuriDomain(?string $ruriDomain = null): static
         return parent::setRuriDomain($ruriDomain);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function setCompany(CompanyInterface $company): static
+    {
+        if (!$this->isNew()) {
+            Assertion::eq(
+                $this->getInitialValue('companyId'),
+                $company->getId()
+            );
+        }
```

Comment:
This is ok but other entities have this implemented using a lifecyle service. See https://github.com/irontec/ivozprovider/blob/bleeding/library/Ivoz/Provider/Domain/Service/FixedCostsRelInvoice/AvoidUpdates.php

I don't know if there are other entites that implement this check in the setter. If these are the first ones, I would prefer to stick to the previous implemented mechanisms and create an AvoidCompanyUpdate.php service for these entities.